contracts/test/utils/deployment/proxy-lib.ts

```diff
@@ -0,0 +1,179 @@
+import {
+  PluginCloneableMockBuild1,
+  PluginCloneableMockBuild1__factory,
+  PluginUUPSUpgradeableMockBuild1,
+  PluginUUPSUpgradeableMockBuild1__factory,
+  ProxyFactory,
+  ProxyFactory__factory,
+} from '../../../typechain';
+import {ProxyCreatedEvent} from '../../../typechain/src/utils/deployment/ProxyFactory';
+import {
+  ADDRESS,
+  findEvent,
+  PROXY_FACTORY_EVENTS,
+} from '@aragon/osx-commons-sdk';
+import {loadFixture} from '@nomicfoundation/hardhat-network-helpers';
+import {SignerWithAddress} from '@nomiclabs/hardhat-ethers/signers';
+import {expect} from 'chai';
+import {ethers} from 'hardhat';
+
+type FixtureResult = {
+  deployer: SignerWithAddress;
+  implementation: PluginUUPSUpgradeableMockBuild1 | PluginCloneableMockBuild1;
+  proxyFactory: ProxyFactory;
+  daoMockAddr: string;
+  initCalldata: string;
+};
+
+async function uupsProxyFixture(): Promise<FixtureResult> {
+  const [deployer] = await ethers.getSigners();
+
+  const implementation = await new PluginUUPSUpgradeableMockBuild1__factory(
+    deployer
+  ).deploy();
+
+  const proxyFactory = await new ProxyFactory__factory(deployer).deploy(
+    implementation.address
+  );
+
+  // Create a mock address with a valid checksum
+  const daoMockAddr = ethers.utils.getAddress(ADDRESS.LAST);
+
+  const initCalldata = implementation.interface.encodeFunctionData(
+    'initialize',
+    [daoMockAddr]
+  );
+
+  return {deployer, implementation, proxyFactory, daoMockAddr, initCalldata};
+}
+
+async function minimalProxyFixture(): Promise<FixtureResult> {
+  const [deployer] = await ethers.getSigners();
+
+  const implementation = await new PluginCloneableMockBuild1__factory(
+    deployer
+  ).deploy();
+
+  const proxyFactory = await new ProxyFactory__factory(deployer).deploy(
+    implementation.address
+  );
+
+  // Create a mock address with a valid checksum
+  const daoMockAddr = ethers.utils.getAddress(ADDRESS.LAST);
+
+  const initCalldata = implementation.interface.encodeFunctionData(
+    'initialize',
+    [daoMockAddr]
+  );
+
+  return {deployer, implementation, proxyFactory, daoMockAddr, initCalldata};
+}
```

Comment:
as in the previous comments about fixtures

Reply:
Done in 35af6a0219c347ad186ea95fc1fb6d290a654599